modules/overrides/limits_test.go

```diff
@@ -0,0 +1,52 @@
+package overrides
+
+import (
+	"encoding/json"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v3"
+)
+
+// Copied from Cortex
+func TestLimitsTagsYamlMatchJson(t *testing.T) {
+	limits := reflect.TypeOf(Limits{})
+	n := limits.NumField()
+	var mismatch []string
+
+	for i := 0; i < n; i++ {
+		field := limits.Field(i)
+
+		// Note that we aren't requiring YAML and JSON tags to match, just that
+		// they either both exist or both don't exist.
+		hasYAMLTag := field.Tag.Get("yaml") != ""
+		hasJSONTag := field.Tag.Get("json") != ""
+
+		if hasYAMLTag != hasJSONTag {
+			mismatch = append(mismatch, field.Name)
+		}
+	}
+
+	assert.Empty(t, mismatch, "expected no mismatched JSON and YAML tags")
+}
+
+// Copied from Cortex and modified
+func TestLimitsStringDurationYamlMatchJson(t *testing.T) {
+	inputYAML := `
+ingestion_rate_strategy: global
```

Comment:
may as well chuck all the fields in here while we're at it.

Reply:
Thanks for this, actually uncovered a bug with the duration type!